content/local/store.go

```diff
@@ -661,6 +660,19 @@ func writeTimestampFile(p string, t time.Time) error {
 	if err != nil {
 		return err
 	}
+	return atomicWrite(p, b, 0666)
+}
 
-	return continuity.AtomicWriteFile(p, b, 0666)
+func atomicWrite(path string, data []byte, mode os.FileMode) error {
+	tmp := fmt.Sprintf("%s.tmp", path)
+	f, err := os.OpenFile(tmp, os.O_RDWR|os.O_CREATE|os.O_EXCL|os.O_SYNC, mode)
```

Comment:
I see the randomness is removed from the filepath. If for some reason this file exists from a previous failure, would `O_TRUNC` be better than `O_EXCL`? Otherwise I see the randomness as unnecssary anyway.

Reply:
updated